src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -257,6 +264,14 @@ Klass* SystemDictionary::handle_resolution_exception(Symbol* class_name,
   return klass;
 }
 
+// Forwards to resolve_or_null
+
+Klass* SystemDictionary::resolve_or_fail(Symbol* class_name, Handle class_loader, Handle protection_domain,
+                                         bool throw_error, TRAPS) {
+  Klass* klass = resolve_or_null(class_name, class_loader, protection_domain, THREAD);
+  // Check for pending exception or null klass, and throw exception
+  return handle_resolution_exception(class_name, throw_error, klass, THREAD);
+}
```

Comment:
How about moving this block of code to its old location to minimize the diff?

Reply:
I moved that resolve_or_fail to be with the rest of the resolve_or_fail, and the resolve_or_nulls.

Reply:
I meant to move the other resolve_or_fail to the header file.